src/analysis/normed_space/star/continuous_functional_calculus.lean

```diff
@@ -0,0 +1,232 @@
+by evaluating `φ` at `a`. This is essentially just evaluation of the `gelfand_transform` of `a`,
+but because we want something in `spectrum ℂ a`, as opposed to
+`spectrum ℂ ⟨a, elemental_star_algebra.self_mem ℂ a⟩` there is slightly more work to do. -/
+@[simps]
+noncomputable def elemental_star_algebra.character_space_to_spectrum (x : A) :
+  C(character_space ℂ (elemental_star_algebra ℂ x), spectrum ℂ x) :=
+{ to_fun := λ φ,
+  { val := φ ⟨x, self_mem ℂ x⟩,
+    property := by simpa only [star_subalgebra.spectrum_eq (elemental_star_algebra.is_closed ℂ x)
+      ⟨x, self_mem ℂ x⟩] using alg_hom.apply_mem_spectrum φ (⟨x, self_mem ℂ x⟩) },
+  continuous_to_fun := continuous_induced_rng.2 (map_continuous $
+    gelfand_transform ℂ (elemental_star_algebra ℂ x) ⟨x, self_mem ℂ x⟩) }
+
+lemma elemental_star_algebra.character_space_to_spectrum_bijective :
+  function.bijective (elemental_star_algebra.character_space_to_spectrum a) :=
+begin
+  refine ⟨λ φ ψ h, star_alg_hom_class_ext ℂ (map_continuous φ) (map_continuous ψ)
+    (by simpa only [elemental_star_algebra.character_space_to_spectrum, subtype.mk_eq_mk,
+      continuous_map.coe_mk] using h), _⟩,
+  rintros ⟨z, hz⟩,
+  set a' : elemental_star_algebra ℂ a := ⟨a, self_mem ℂ a⟩,
+  rw [(show a = a', from rfl), ←star_subalgebra.spectrum_eq
+    (elemental_star_algebra.is_closed ℂ a) a', ←spectrum.gelfand_transform_eq a',
+    continuous_map.spectrum_eq_range] at hz,
+  obtain ⟨φ, rfl⟩ := hz,
+  exact ⟨φ, rfl⟩,
+end
+
+/-- The homeomorphism between the character space of the unital C⋆-subalgebra generated by a
+single normal element `a : A` and `spectrum ℂ a`. -/
+noncomputable def elemental_star_algebra.character_space_homeo :
+  character_space ℂ (elemental_star_algebra ℂ a) ≃ₜ spectrum ℂ a :=
+@continuous.homeo_of_equiv_compact_to_t2 _ _ _ _ _ _
+  (equiv.of_bijective (elemental_star_algebra.character_space_to_spectrum a)
+    (elemental_star_algebra.character_space_to_spectrum_bijective a))
+    (map_continuous $ elemental_star_algebra.character_space_to_spectrum a)
+
+/-- **Continuous functional calculus.** Given a normal element `a : A` of a unital C⋆-algebra,
+the continuous functional calculus is a `star_alg_equiv` from the complex-valued continuous
+functions on the spectrum of `a` to the unital C⋆-subalgebra generated by `a`. Moreover, this
+equivalence identifies `(continuous_map.id ℂ).restrict (spectrum ℂ a))` with `a`; see
+`continuous_functional_calculus_map_id`. As such it extends the polynomial functional calculus. -/
+noncomputable def continuous_functional_calculus :
+  C(spectrum ℂ a, ℂ) ≃⋆ₐ[ℂ] elemental_star_algebra ℂ a :=
+((elemental_star_algebra.character_space_homeo a).comp_star_alg_equiv' ℂ ℂ).trans
+  (gelfand_star_transform (elemental_star_algebra ℂ a)).symm
+
+lemma continuous_functional_calculus_map_id :
+  continuous_functional_calculus a ((continuous_map.id ℂ).restrict (spectrum ℂ a)) =
+    ⟨a, self_mem ℂ a⟩ :=
+star_alg_equiv.symm_apply_apply _ _
```

Comment:
can you also write down the lemma that for the evaluation of a polynomial, you get what you expect?

Reply:
If you don't mind, I would like to avoid this for now. Reason: although this is the most *precise* version of the continuous functional calculus, I don't think it's the one we're going to end up using in practice (even though those will use this one under the hood). Frédéric and I have started discussing what a general API might look like. I suspect that we want something to the effect of a type class for continuous functional calculi (because at the very least we'll want a version for selfadjoint elements that uses real-valued functions instead) that provides a ⋆-algebra homomorphism from the continuous functions (maybe on the spectrum? but maybe not?) into the algebra `A` which sends the identity map to `a` (and is continuous; redundant for C⋆-algebras). These conditions uniquely specify the ⋆-homomorphism because it specifies the action on polynomials and by Stone-Weierstrass these are dense in the continuous functions, so continuity makes it unique.

Anyway, the point is, I think we'll want these sorts of theorem more generically for the API we eventually develop, so I don't think we want to include them now. It would be nice to avoid the `elemental_star_subalgebra` subtype whenever possible.